include/cglm/aabb2d.h

```diff
@@ -0,0 +1,279 @@
+
+  /* translation + min(xa, xb) + min(ya, yb) + min(za, zb) */
+  glm_vec2(m[3], v[0]);
+  glm_vec2_minadd(xa, xb, v[0]);
+  glm_vec2_minadd(ya, yb, v[0]);
+  glm_vec2_minadd(za, zb, v[0]);
+
+  /* translation + max(xa, xb) + max(ya, yb) + max(za, zb) */
+  glm_vec2(m[3], v[1]);
+  glm_vec2_maxadd(xa, xb, v[1]);
+  glm_vec2_maxadd(ya, yb, v[1]);
+  glm_vec2_maxadd(za, zb, v[1]);
+
+  glm_vec2_copy(v[0], dest[0]);
+  glm_vec2_copy(v[1], dest[1]);
+}
+
+/*!
+ * @brief merges two AABB bounding aabb and creates new one
+ *
+ * two aabb must be in same space, if one of aabb is in different space then
+ * you should consider to convert it's space by glm_aabb_space
+ *
+ * @param[in]  aabb1 bounding aabb 1
+ * @param[in]  aabb2 bounding aabb 2
+ * @param[out] dest merged bounding aabb
+ */
+CGLM_INLINE
+void
+glm_aabb2d_merge(vec2 aabb1[2], vec2 aabb2[2], vec2 dest[2]) {
+  dest[0][0] = glm_min(aabb1[0][0], aabb2[0][0]);
+  dest[0][1] = glm_min(aabb1[0][1], aabb2[0][1]);
+
+  dest[1][0] = glm_max(aabb1[1][0], aabb2[1][0]);
+  dest[1][1] = glm_max(aabb1[1][1], aabb2[1][1]);
+}
+
+/*!
+ * @brief crops a bounding aabb with another one.
+ *
+ * this could be useful for gettng a baabb which fits with view frustum and
+ * object bounding aabbes. In this case you crop view frustum aabb with objects
+ * aabb
+ *
+ * @param[in]  aabb     bounding aabb 1
+ * @param[in]  cropaabb crop aabb
+ * @param[out] dest    cropped bounding aabb
+ */
+CGLM_INLINE
+void
+glm_aabb2d_crop(vec2 aabb[2], vec2 cropAabb[2], vec2 dest[2]) {
+  dest[0][0] = glm_max(aabb[0][0], cropAabb[0][0]);
+  dest[0][1] = glm_max(aabb[0][1], cropAabb[0][1]);
+
+  dest[1][0] = glm_min(aabb[1][0], cropAabb[1][0]);
+  dest[1][1] = glm_min(aabb[1][1], cropAabb[1][1]);
+}
+
+/*!
+ * @brief crops a bounding aabb with another one.
+ *
+ * this could be useful for gettng a baabb which fits with view frustum and
+ * object bounding aabbes. In this case you crop view frustum aabb with objects
+ * aabb
+ *
+ * @param[in]  aabb      bounding aabb
+ * @param[in]  cropaabb  crop aabb
+ * @param[in]  clampaabb miniumum aabb
+ * @param[out] dest     cropped bounding aabb
+ */
+CGLM_INLINE
+void
+glm_aabb2d_crop_until(vec2 aabb[2],
+                    vec2 cropAabb[2],
+                    vec2 clampAabb[2],
+                    vec2 dest[2]) {
+  glm_aabb2d_crop(aabb, cropAabb, dest);
+  glm_aabb2d_merge(clampAabb, dest, dest);
+}
+
+/*!
+ * @brief check if AABB intersects with frustum planes
+ *
+ * this could be useful for frustum culling using AABB.
+ *
+ * OPTIMIZATION HINT:
+ *  if planes order is similar to LEFT, RIGHT, BOTTOM, TOP, NEAR, FAR
+ *  then this method should run even faster because it would only use two
+ *  planes if object is not inside the two planes
+ *  fortunately cglm extracts planes as this order! just pass what you got!
+ *
+ * @param[in]  aabb     bounding aabb
+ * @param[in]  planes  frustum planes
+ */
+CGLM_INLINE
+bool
+glm_aabb2d_frustum(vec2 aabb[2], vec4 planes[6]) {
```

Comment:
if you dont use this, maybe we should remove for now and bring back when it is well defined. Because frustum is 3D, since there is no Z here intersection could only be done two axes e.g. far & near planes or some of them ... 

Any ideas?